connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -582,4 +626,161 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
             messageBatch.clear();
         }
     }
+
+    static class SinkTaskMetricsGroup {
+        private final MetricGroup metricGroup;
+        private final Sensor sinkRecordRead;
+        private final Sensor sinkRecordSend;
+        private final Sensor sinkRecordLag;
+        private final Sensor partitionCount;
+        private final Sensor offsetSeqNum;
+        private final Sensor offsetCompletion;
+        private final Sensor offsetCompletionSkip;
+        private final Sensor putBatchTime;
+        private final AtomicReference<Map<TopicPartition, OffsetAndMetadata>> consumedOffsets = new AtomicReference<>();
```

Comment:
Do these need to be `AtomicReference`s? What are the threads accessing it? Seems like the main processing thread for the sink task would be the only one?

Reply:
No, not anymore. I used to have the thread reading the metrics also accessing these and computing the lag on-the fly, but that wasn't correct. I'll remove the AtomicReferences.